mesonbuild/interpreterbase/interpreterbase.py

```diff
@@ -168,21 +168,18 @@ def evaluate_codeblock(self, node: mparser.CodeBlockNode, start: int = 0, end: T
         if node is None:
             return
         if not isinstance(node, mparser.CodeBlockNode):
-            e = InvalidCode('Tried to execute a non-codeblock. Possibly a bug in the parser.')
-            e.lineno = node.lineno
-            e.colno = node.colno
-            raise e
+            raise InvalidCode.from_node('Tried to execute a non-codeblock. Possibly a bug in the parser.', node=node)
         statements = node.lines[start:end]
         i = 0
         while i < len(statements):
             cur = statements[i]
+            self.current_lineno = cur.lineno
             try:
-                self.current_lineno = cur.lineno
                 self.evaluate_statement(cur)
             except MesonException as e:
                 # Assume that if file is set, then the information in the exception is correct
                 if e.file is None:
-                    e.set_from_node(cur)
+                    e.set_from_node(self.current_node)
```

Comment:
oops, I squashed it wrong then

Reply:
This code really suffers (for ease of understanding) from the fact that we have `cur` at all... because it's not structurally clear the `cur` and `self.current_node` may be different after we call `self.evaluate_statement` I could re-name the variable, but that doesn't *really* help unless I name it something awful like `pass_this_to_evaluate_statement_and_use_self_current_node_instead`... Since it seems I'm not the first person to run into this issue.

Reply:
Maybe "start_node"?

Reply:
I think "cur" was originally named that way because it's the root of the node that the current interpreter context is pointing at... which is of course very different from the node we're at if and when we traceback.

Naming it something like "start" feels to me like it conveys the understanding that it's the node where we started at the time we entered the function/loop, but isn't necessarily where we ended up.